src/main/java/address/model/ModelManager.java

```diff
@@ -147,27 +143,16 @@ public void clearModel() {
      * Adds a person to the model
      * @throws DuplicatePersonException when this operation would cause duplicates
      */
-    public synchronized void addPerson(ReadOnlyPerson personToAdd) throws DuplicatePersonException {
-        final Person toAdd = new Person(personToAdd);
-        if (backingPersonList().contains(toAdd)) {
-            throw new DuplicatePersonException(personToAdd);
-        }
+    public synchronized void addPerson(ReadOnlyPerson data) throws DuplicatePersonException {
+        Person toAdd;
+        do { // make sure no id clashes.
+            toAdd = new Person(Math.abs(UUID.randomUUID().hashCode()));
+        } while (backingPersonList().contains(toAdd));
+        toAdd.update(data);
```

Comment:
This is a temporary workaround for assigning unique ids, will be replaced by a proper id assigning implementation in the cloud component